configure.ac

```diff
@@ -96,6 +96,16 @@ AS_IF([test -n "$missing_libevent"],
       [AC_MSG_ERROR([could not find the libevent libraries])])
 LIBS="$save_LIBS"
 
+# TCMalloc gubbins
+AC_ARG_WITH([tcmalloc],
+            [AS_HELP_STRING([--without-tcmalloc],
+                            [disable tcmalloc for memory allocations])],
+            [],
+            [with_tcmalloc=yes])
+AS_IF([test "x$with_tcmalloc" != xno],
+      [AC_CHECK_LIB([tcmalloc], [malloc],,
+                    [AC_MSG_FAILURE([no tcmalloc found (use --without-tcmalloc to disable)])])])
```

Comment:
If we think that it's a good idea to run with tcmalloc, then it seems like we should encourage it - leaving it up to someone to remember to set `LIBS` to correctly in order to link in something which we recommend but which they won't be able to easily tell when/if they forget to do so doesn't seem like a good idea.

Note that this is a `--without-foobar` flag for the "exceptional" cases where it's a PITA/doesn't-work-on-my-platform scenarios.

Do you have a good example of why you're not in favour of these types of things?